fuzzers/patch_parse_fuzzer.c

```diff
@@ -0,0 +1,38 @@
+/*
+ * libgit2 patch parser fuzzer target.
+ *
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+
+#include "git2.h"
+#include "patch.h"
+#include "patch_parse.h"
+
+#define UNUSED(x) (void)(x)
+
+int LLVMFuzzerInitialize(int *argc, char ***argv)
+{
+	UNUSED(argc);
+	UNUSED(argv);
+
+	if (git_libgit2_init() < 0)
+		abort();
+
+	return 0;
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+	if (size) {
+		git_patch *patch = NULL;
+		git_patch_options opts = GIT_PATCH_OPTIONS_INIT;
+		opts.prefix_len = (uint32_t)data[0];
```

Comment:
Sorry, I somehow didn't see this in my first round, but I'm wondering about the first byte of `data` here. So you're skipping the first byte and use it as `prefix_len`, right? But judging from the fuzzing corpora, the first byte is simply part of the diff, and thus it is _very_ likely that the initial line is a non-valid diff header already. So shouldn't we just leave the `prefix_len` alone and just use `data` as input for `git_patch_from_buffer` directly?

E.g. just:

```
git_patch *patch = NULL;
git_patch_from_buffer(&patch, data, size, NULL);
git_patch_free(patch);
```